### PR TITLE
install gtp2ogs and node_modules locally (no global -g), default to devel, add package-lock.json

### PR DESCRIPTION
**Pull request by wonderingabout**
edit: rebased with no change since the last few days
rewrote commit notes cleanly

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -2,179 +2,128 @@
...
+After that, it is mandatory to also yout AI settings. To do that, add a `--`
+ (with one space before and one space after the `--` ), to separate `<gtp2ogsarguments>`
+ from your bot path and `<botarguments>`.
 
-### 4. Most common usage: start gtp2ogs.js using nodejs
+## Beta OGS
 
-For linux (preferably as sudo):
+Before you can play on official [OGS server](https://online-go.com/) server, OGS admins
+ will often ask that you run tests to make sure everything is running fine.
```
Comment (sanderland) [inline README.md:62]: I did not find this to be true

Comment (wonderingabout) [reply README.md:62]: indeed they dont always ask it, but they often do, especially if they see the new bot admin is fairly new
so its indeed not always true but often

File: README.md [hunk 2]
```diff
@@ -2,179 +2,128 @@
...
-```
+Then, choose the wanted `<gtp2ogsarguments>` from the above list of all options section
+ (no need to use all these options, only the ones you want), and separate each option with a space.
 
-For details or help, you can see:
+note: some gtp2ogsarguments have default so they are enabled even if you don't specify them.
 
-- for linux: [3A3) Recommended: Upgrade gtp2ogs.js from old branch to “devel” branch (latest)](https://github.com/wonderingabout/gtp2ogs-tutorial/blob/master/docs/3A3-linux-optional-upgrade-to-devel.md)
-- for windows: [3B3) Recommended: Upgrade gtp2ogs from old branch to devel (latest) branch](https://github.com/wonderingabout/gtp2ogs-tutorial/blob/master/docs/3B3-windows-optional-upgrade-to-devel.md)
+After that, it is mandatory to also yout AI settings. To do that, add a `--`
```
Comment (sanderland) [inline README.md:55]: this sentence is a word

Comment (sanderland) [reply README.md:55]: and I would rewrite in terms of 'the command and arguments used to run your bot' rather than talking about settings (which could be anywhere)

File: README.md [hunk 3]
```diff
@@ -2,179 +2,128 @@
...
-```Shell
-node /usr/lib/node_modules/gtp2ogs/gtp2ogs.js --username <yourbotusername> --apikey <apikey> <gtp2ogsargument1> <gtp2ogsargument2> -- /replace/with/full/path/to/your/ai/runfile.file <botargument1> <botargument2>
-```
+You need to do these tests in [beta OGS server](https://beta.online-go.com/).
 
-For windows (preferably as admin):
+When all testing is OK, you can remove the `--beta` and play in the real OGS.
 
-```Shell
-pushd C:\Program Files\nodejs && node.exe C:\Users\yourwindowsusername\AppData\Roaming\npm\node_modules\gtp2ogs\gtp2ogs.js --username <yourbotusername> --apikey <apikey> <gtp2ogsargument1> <gtp2ogsargument2> -- C:\Users\path\to\your\ai\executable.exe <botargument1> <botargument2>
```
Comment (sanderland) [inline README.md:113]: where does pushd come from?

File: README.md [hunk 4]
```diff
@@ -2,179 +2,128 @@
...
+### For Windows
 
-You can find a list of all possible to use gtp2ogs arguments [here](/docs/OPTIONS-LIST.md).
+General use case is something like this:
 
-Before putting `<gtp2ogsarguments>`, you have to put the options [username](https://github.com/online-go/gtp2ogs/blob/devel/docs/OPTIONS-LIST.md#username)
-and [apikey](https://github.com/online-go/gtp2ogs/blob/devel/docs/OPTIONS-LIST.md#apikey)
-first.
+`
+pushd C:\Program Files\nodejs\node.exe C:\Users\yourusername\gtp2ogs-node\node_modules\gtp2ogs\gtp2ogs.js --username <yourbotusername> --apikey <apikey> <gtp2ogsargument1> <gtp2ogsargument2> -- C:\Users\path\to\your\ai\executable.exe <botargument1> <botargument2>
```
Comment (sanderland) [inline README.md:79]: i would make these example command as small as possible

File: README.md [hunk 5]
```diff
@@ -2,179 +2,128 @@
...
-To avoid accidental misclicks while spectating a game from
-your bot account,
+## Submit Move Button
 
-see: [Submit-Move Button](https://github.com/wonderingabout/gtp2ogs-tutorial#important-submit-move-button)
+To avoid accidental misclicks while spectating a game from your bot account, see
+ [Submit-Move Button](https://github.com/wonderingabout/gtp2ogs-tutorial#important-submit-move-button).
 
-### Show winrate and variations ingame
+## Show winrate and variations ingame
```
Comment (sanderland) [inline README.md:107]: why is this a section here at all, rather than somewhere in options?

File: README.md [hunk 6]
```diff
@@ -2,179 +2,128 @@
...
-interactive chat, or ask simple and quick questions (github Issues are
-preferred for long problems because they live a track that can be useful
-later)!
+ interactive chat, or ask simple and quick questions, however github issues
+ are preferred for long problems because they live a track that can be useful
+ later)
+
+## Developer Notes
+
+You can find notes for developers [in DEV.md](/docs/DEV.md).
```
Comment (sanderland) [inline README.md:129]: [here](/docs/DEV.md)

File: docs/INSTALL-GUIDE-LINUX.md [hunk 7]
```diff
@@ -0,0 +1,80 @@
...
+
+## 1. install node.js and npm
+
+It is recommend to use [nvm (Node Version Manager)](https://github.com/nvm-sh/nvm#install--update-script),
+ use this link to install latest nvm version.
+
+For example:
+
+```Shell
+curl -o- https://raw.githubusercontent.com/nvm-sh/nvm/v0.35.3/install.sh | bash
```
Comment (sanderland) [inline docs/INSTALL-GUIDE-LINUX.md:15]: can this not be done with apt-get or something?

File: docs/INSTALL-GUIDE-LINUX.md [hunk 8]
```diff
@@ -0,0 +1,80 @@
...
+
+This step is **highly recommended**.
+
+By default, npm installs an old branch of gtp2ogs that does not include latest
+ improvements, new features, and fixes.
+
+You can go to `~/gtp2ogs-node/node_modules/gtp2ogs`, delete all existing
+ files (make a copy backup just in case), then either:
+
+- download ZIP from github [the devel branch](https://github.com/online-go/gtp2ogs/tree/devel),
```
Comment (sanderland) [inline docs/INSTALL-GUIDE-LINUX.md:66]: why not git clone?

File: docs/INSTALL-GUIDE-WINDOWS.md [hunk 9]
```diff
@@ -0,0 +1,55 @@
...
+By default, npm installs an old branch of gtp2ogs that does not include latest
+ improvements, new features, and fixes.
+
+You can go to `C:\Users\yourusername\gtp2ogs-node\node_modules\gtp2ogs`, delete all existing
+ files (make a copy backup just in case), then either:
+
+- download ZIP from github [the devel branch](https://github.com/online-go/gtp2ogs/tree/devel),
+ then go inside `C:\Users\yourusername\gtp2ogs-node\node_modules\gtp2ogs` folder,
+ and paste all the new devel branch files.
+- or another use `git` if you're familiar with it to install devel from online-go/gtp2ogs or
```
Comment (sanderland) [inline docs/INSTALL-GUIDE-WINDOWS.md:44]: typos

File: README.md (hunk 2 above)
Comment (wonderingabout) [reply README.md:55]: indeed, the wording is wrong also :+1:

File: README.md (hunk 5 above)
Comment (wonderingabout) [reply README.md:107]: because it's an important add to gtp2ogs, so i think it's worth giving it extra visibility (just a link to the option details)

File: README.md (hunk 6 above)
Comment (wonderingabout) [reply README.md:129]: i dont understand, the link seems to correctly redirect to `/docs/DEV.md` on my local branch

File: docs/INSTALL-GUIDE-LINUX.md (hunk 7 above)
Comment (wonderingabout) [reply docs/INSTALL-GUIDE-LINUX.md:15]: no, apt only provides nodejs 10.19.xx (it was the old method by the way) default, and if we add the curl from nodejs repo we get package conflicts and broken dependencies

Comment (wonderingabout) [reply docs/INSTALL-GUIDE-LINUX.md:15]: i added a note for it to be clear

File: docs/INSTALL-GUIDE-LINUX.md (hunk 8 above)
Comment (wonderingabout) [reply docs/INSTALL-GUIDE-LINUX.md:66]: it is possible to use git (as explained below), but for beginners they may find it easier to copy paste the files directly

Comment (wonderingabout) [reply docs/INSTALL-GUIDE-LINUX.md:66]: but indeed i forgot to add the git clone command line, added :+1:

File: docs/INSTALL-GUIDE-WINDOWS.md (hunk 9 above)
Comment (wonderingabout) [reply docs/INSTALL-GUIDE-WINDOWS.md:44]: fixed, thanks :+1:

File: README.md (hunk 3 above)
Comment (wonderingabout) [reply README.md:113]: i dont understand this question

File: README.md (hunk 4 above)
Comment (wonderingabout) [reply README.md]: i removed a few commands (bans (since we dont always use these), maxconnectedgames and maxconnectedgamesperuser (since we provide defaults for these and are unlikely to want to change the defaults), i also added ogspv to highlight that it exists

for the other options i thought it would be a good way to illustrate a real use case of gtp2ogs to show the range of possibilities to a new user

**Comment by wonderingabout** [comment]
> some suggestions

> some suggestions

big thanks, they are for the most really accurate and relevant, thanks! :100: 

feel free to add other feedback

File: README.md (hunk 6 above)
Comment (sanderland) [reply README.md:121]: I mean, link with the word 'here'

Comment (wonderingabout) [reply README.md]: ah i see, fixed :+1:

File: README.md [hunk 10]
```diff
@@ -2,179 +2,112 @@
...
 
-For linux (preferably as sudo):
+- [--username](/docs/OPTIONS-LIST.md#username)
+- [--apikey](/docs/OPTIONS-LIST.md#apikey)
 
-```Shell
-node /usr/lib/node_modules/gtp2ogs/gtp2ogs.js --username <yourbotusername> --apikey <apikey> <gtp2ogsargument1> <gtp2ogsargument2> -- /replace/with/full/path/to/your/ai/runfile.file <botargument1> <botargument2>
-```
+Then, choose the wanted `options` from the above list of all options section
+ (no need to use all these options, only the ones you want), and separate each option with a space.
```
Comment (Dorus) [inline README.md:51]: It might be better to explain options can be set with command line arguments, and those have to be separated by spaces. This explanation could also come earlier in the introduction i think.

I'm not even sure you have to mention argument have to be space separated, this is pretty standard. Just mention arguments can be used to set the various options.

File: README.md [hunk 11]
```diff
@@ -2,179 +2,112 @@
...
 
-For windows (preferably as admin):
+note: some options have default so they are enabled even if you don't specify them.
 
-```Shell
-pushd C:\Program Files\nodejs && node.exe C:\Users\yourwindowsusername\AppData\Roaming\npm\node_modules\gtp2ogs\gtp2ogs.js --username <yourbotusername> --apikey <apikey> <gtp2ogsargument1> <gtp2ogsargument2> -- C:\Users\path\to\your\ai\executable.exe <botargument1> <botargument2>
-```
+After that, specify your bot command (and its arguments) using a , add a ` -- `
```
Comment (Dorus) [inline README.md:55]: > After that, specify your bot command (and its arguments) using a , add a ` -- `

I'm not sure what you mean here, there is no need for a comma in there, i think the words ` using a ` have to be removed here.

File: README.md [hunk 12]
```diff
@@ -2,179 +2,112 @@
...
+ (with one space before and one space after the `--` ), to separate `options`
+ from your bot path and `bot arguments`.
```
Comment (Dorus) [inline README.md:57]: Why is bot arguments in code style?

Did you mean to articulate it? I might prefer italic `from your *bot path* and *bot arguments*.`
>  from your *bot path* and *bot arguments*.

File: README.md [hunk 13]
```diff
@@ -2,179 +2,112 @@
...
-arguments from your bot arguments
+All in all, general use is something like this:
 
-note 2: the number of `<gtp2ogsarguments>` and `<botarguments>` is not limited,
-here only 2 were shown but it possible to use for example 3,4,5 , or as many as
-you want
-  
-note 3: to play on [beta OGS server](https://beta.online-go.com/) instead of the
-[OGS server](https://online-go.com/), add the `--beta` argument
+`path/to/node_executable path/to/gtp2ogs.js --username <yourbotusername> --apikey <apikey> --option1 <argument1> --option2 <argument2> -- path/to/your/ai/executable <botoption1> <botargument1> <botoption2> <botargument2>`
```
Comment (Dorus) [inline README.md:61]: You should probably make this a code block, it's too long for inline style.

\`\`\`Shell
[code]
\`\`\`

File: README.md [hunk 14]
```diff
@@ -2,179 +2,112 @@
...
 
-Then, choose the wanted `<gtp2ogsarguments>` from the above [list of all options](/docs/OPTIONS-LIST.md) section (no need to use all these options, only the ones you want),
-and separate each option with a space
+### For Windows
 
-After that, add a `--` (with one space before and one space after the `--` ),
-to separate `<gtp2ogsarguments>` from your bot path and `<botarguments>`, as
-shown in
-[Most common usage earlier](#4-most-common-usage--start-gtp2ogsjs-using-nodejs)
+`C:\Program Files\nodejs\node.exe C:\Users\yourusername\gtp2ogs-node\node_modules\gtp2ogs\gtp2ogs.js --apikey 5rg46ze84f68e4g6v4e4g68es4g8 --username somebot --debug --ogspv PhoenixGo --noclock --unrankedonly --minrank 1d --noautohandicapranked --maxhandicap 0 --boardsizes 19 --komis 7.5 --speeds live --maxmaintimelive 600 --maxperiodsliveranked 5 --maxperiodsliveunranked 10 --maxperiodtimeliveranked 60 --maxperiodtimeliveunranked 180 -- C:\Users\yourusername\PhoenixGo\mcts\mcts_main --gtp --config_path C:\Users\yourusername\PhoenixGo\etc\mcts_1gpu_beta.conf --logtostderr --v 1`
```
Comment (Dorus) [inline README.md:71]: No need to use `C:\Program Files\nodejs\` if you installed node.js and have it in your path.

File: README.md (hunk 13 above)
Comment (Dorus) [reply README.md:61]: This also count for the other blocks below.

File: README.md [hunk 15]
```diff
@@ -2,179 +2,112 @@
...
+ Being a bot admin on OGS means:
 
 - you can put your bot online whenever you want
 - choose settings to automatically accept or reject challenges
-(example: rank, boardsize, blitz/live/correspondence, etc.), or to
-control your GPU/computing device power spent (max number of
-simultaneous games, corrqueue, etc.)
-- choose your bot-specific options (for example `-v 1600` for leela-zero,
-etc.)
+ (example: rank, boardsize, blitz/live/correspondence, etc.), or to
```
Comment (Dorus) [inline README.md:16]: board size

File: README.md [hunk 16]
```diff
@@ -2,179 +2,112 @@
...
 
-see: [Submit-Move Button](https://github.com/wonderingabout/gtp2ogs-tutorial#important-submit-move-button)
+## Submit Move Button
 
-### Show winrate and variations ingame
+To avoid accidental misclicks while spectating a game from your bot account, see
+ [Submit-Move Button](https://github.com/wonderingabout/gtp2ogs-tutorial#important-submit-move-button).
 
-**gtp2ogs has default support for showing ingame winrate and variations for some AI.**
+## Show winrate and variations ingame
```
Comment (Dorus) [inline README.md:92]: in game
or
during games

File: README.md [hunk 17]
```diff
@@ -2,179 +2,112 @@
...
 
-see: [--ogspv](/docs/OPTIONS-LIST.md/#ogspv)
+gtp2ogs has default support for showing ingame winrate and variations for some AI,
+ see: [--ogspv](/docs/OPTIONS-LIST.md/#ogspv)
 
 ## Community Involvment
```
Comment (Dorus) [inline README.md:97]: Involvement

File: README.md [hunk 18]
```diff
@@ -2,179 +2,112 @@
...
 
-- You may be interested in [Contributing](/docs/CONTRIBUTING.md)
 - You may also use [Issues](https://github.com/online-go/gtp2ogs/issues)
-to report issues.
-
-- You may be interested in joining the Discord chat:
-In leela zero discord (`#bots` channel): <https://discord.gg/HZ23Cp9>
+ to report issues.
+- Contributing in most welcome, if you want to add features or submit fixes
```
Comment (Dorus) [inline README.md:101]: Shouldn't that be `is most welcome`?

File: docs/INSTALL-GUIDE-LINUX.md [hunk 19]
```diff
@@ -0,0 +1,84 @@
...
+Then go inside this folder and install gtp2ogs with npm:
+
+```Shell
+cd ~/gtp2ogs-node
+npm install gtp2ogs
+```
+
+gtp2ogs will be installed in `~/gtp2ogs-node/node_modules/gtp2ogs`
+
+## 3. Recommended: Upgrade to devel branch
```
Comment (Dorus) [inline docs/INSTALL-GUIDE-LINUX.md:60]: Why first download old gtp2ogs if you want to switch to dev anyway? I would skip step 2 if you are planing to do step 3 anyway.
That would also eliminate the need to delete the `node_modules/gtp2ogs` folder at this step.

File: docs/INSTALL-GUIDE-LINUX.md [hunk 20]
```diff
@@ -0,0 +1,84 @@
...
+
+This step is **highly recommended**.
```
Comment (Dorus) [inline docs/INSTALL-GUIDE-LINUX.md:62]: Note we're recommending people to switch to a beta version here.

Not per sé bad, it has a lot of new good features, but we should probably put a note about that here.

Also makes me think if we should add some branching strategy where users can use `master` and we work in `develop` branch or so. Only merging to master once we know it's stable for a bit.

Comment (Dorus) [reply docs/INSTALL-GUIDE-LINUX.md:62]: ps. this comment also count for the windows guide.

File: README.md [hunk 21]
```diff
@@ -2,179 +2,112 @@
 
 [![Build Status](https://travis-ci.org/online-go/gtp2ogs.svg?branch=devel)](https://travis-ci.org/online-go/gtp2ogs)
 
-This javascript tool allows all bots/AI that support
-[GTP (Go Text Protocol)](https://senseis.xmp.net/?GoTextProtocol)
-to communicate with [OGS (Online-Go.com Server)](https://online-go.com/).
+ This javascript tool allows all bots/AI that support
```
[the remark this answers is not in the document]
Comment (wonderingabout) [reply README.md:5]: the spaces on lines 5,9,12 are indeed typos, the other ones at the beginning of a newline are part of markdownlint style

File: README.md [hunk 22]
```diff
@@ -2,179 +2,112 @@
...
+ control your GPU/computing device power spent (max number of
+ simultaneous games, corrqueue, etc.)
+- choose your bot-specific options (for example `-v 1600` for leela-zero, etc.)
 
 ## Full tutorial
 
 [for windows and linux, with screenshots, examples, and detailed explanations](https://github.com/wonderingabout/gtp2ogs-tutorial)
 
-## Quick Guide
+## Quick Install Guide
```
[the remark this answers is not in the document]
Comment (wonderingabout) [reply README.md:27]: indeed, it seems a more accurate word! https://wiki.archlinux.org/index.php/installation_guide

File: README.md [hunk 23]
```diff
@@ -2,179 +2,112 @@
...
 
-### 1. install node.js and npm
+Installing node.js will also install:
```
[the remark this answers is not in the document]
Comment (wonderingabout) [reply README.md:29]: yes, how about this `The "node.js" package is bundled with:` ?

File: README.md [hunk 24]
```diff
@@ -2,179 +2,112 @@
...
-
-```Shell
-cd /usr/lib/node_modules/gtp2ogs/
-sudo npm install
-```
-
-- for windows:
-
-Open a node.js command prompt as admin, then:
+## Options
```
[the remark this answers is not in the document]
Comment (wonderingabout) [reply README.md:39]: fixed

File: README.md [hunk 25]
```diff
@@ -2,179 +2,112 @@
...
 
-For details or help, you can see:
+You can find the full list of all existing gtp2ogs options [here](/docs/OPTIONS-LIST.md).
 
-- for linux: [3A3) Recommended: Upgrade gtp2ogs.js from old branch to “devel” branch (latest)](https://github.com/wonderingabout/gtp2ogs-tutorial/blob/master/docs/3A3-linux-optional-upgrade-to-devel.md)
-- for windows: [3B3) Recommended: Upgrade gtp2ogs from old branch to devel (latest) branch](https://github.com/wonderingabout/gtp2ogs-tutorial/blob/master/docs/3B3-windows-optional-upgrade-to-devel.md)
+### Options settings
 
-### 4. Most common usage: start gtp2ogs.js using nodejs
+Before putting `options`, you have to first put the mandatory options:
```
[the remark this answers is not in the document]
Comment (wonderingabout) [reply README.md]: yes, i thought the same but forgot, thanks

File: README.md [hunk 26]
```diff
@@ -2,179 +2,112 @@
...
 
-For linux (preferably as sudo):
+- [--username](/docs/OPTIONS-LIST.md#username)
+- [--apikey](/docs/OPTIONS-LIST.md#apikey)
 
-```Shell
-node /usr/lib/node_modules/gtp2ogs/gtp2ogs.js --username <yourbotusername> --apikey <apikey> <gtp2ogsargument1> <gtp2ogsargument2> -- /replace/with/full/path/to/your/ai/runfile.file <botargument1> <botargument2>
-```
+Then, choose the wanted `options` from the above list of all options section
```
[the remark this answers is not in the document]
Comment (wonderingabout) [reply README.md]: actually i simpified it now and moved the options-list link where we actually need it

chronological and logical :100:

File: README.md (hunk 10 above)
Comment (wonderingabout) [reply README.md]: exactly what i did!

then i'll also remove it, it is obvious from the examples anyways so better not overload the readme

File: README.md (hunk 11 above)
Comment (wonderingabout) [reply README.md]: yes it was very messy, fixed

File: README.md (hunk 12 above)
Comment (wonderingabout) [reply README.md]: actually i wrote it in normal text

File: README.md (hunk 13 above)
Comment (wonderingabout) [reply README.md]: it was a code block at first, but then i thought it was not clearly visible in one glance

isnt it better to be able to see it all without scrolling?

File: README.md (hunk 14 above)
Comment (wonderingabout) [reply README.md:81]: then just node.exe ?

File: README.md (hunk 16 above)
Comment (wonderingabout) [reply README.md:102]: in games?

File: README.md (hunk 17 above)
Comment (wonderingabout) [reply README.md:107]: unrelated but it's actually not a default (not auto enabled) but a "native" feature

fixed typo too

File: README.md (hunk 18 above)
Comment (wonderingabout) [reply README.md:111]: yes, fixed typo

File: docs/INSTALL-GUIDE-LINUX.md (hunk 19 above)
Comment (wonderingabout) [reply docs/INSTALL-GUIDE-LINUX.md:59]: yes!!!!

but how do you install latest devel with npm ?
(for linux and windows)

File: docs/INSTALL-GUIDE-LINUX.md (hunk 20 above)
Comment (wonderingabout) [reply docs/INSTALL-GUIDE-LINUX.md:61]: yes, i'll explain that it is beta

can you explain the branching strategy in detail?

Comment (wonderingabout) [reply docs/INSTALL-GUIDE-LINUX.md]: is it how to do it ?

Comment (wonderingabout) [reply docs/INSTALL-GUIDE-LINUX.md]: is it how to do it?
https://stackoverflow.com/a/39732501

for example?
```Shell
npm install git://github.com/online-go/gtp2ogs.git#devel --save
```

Comment (wonderingabout) [reply docs/INSTALL-GUIDE-LINUX.md]: oh wow it works!!

npm install devel

its not a git repo so we cant clone it, but at least we have latest devel natively

Comment (wonderingabout) [reply docs/INSTALL-GUIDE-LINUX.md]: found an even better way (now we clone from github and can use git in the repo), but we need to add node_modules and package-lock.json to git ignore

```Shell
git clone -b devel https://github.com/online-go/gtp2ogs && cd gtp2ogs && npm i
```

npm install clone

from: https://stackoverflow.com/a/45847287

File: .gitignore [hunk 27]
```diff
@@ -36,5 +36,6 @@ nosetests.xml
 .pydevproject
 
 node_modules
+package-lock.json
```
Comment (Dorus) [inline .gitignore:39]: > This file is intended to be committed into source repositories

https://docs.npmjs.com/configuring-npm/package-lock-json.html

Comment (wonderingabout) [reply .gitignore:39]: i see, then i should remove it again i guess, thanks

Comment (wonderingabout) [reply .gitignore]: fixed

Comment (Dorus) [reply .gitignore]: Yes, and once you've done a clean `npm install`, you should also commit this file.

**Comment by roy7** [comment]
Any new or outstanding issues before merge? @Dorus @sanderland

File: .gitignore (hunk 27 above)
Comment (wonderingabout) [reply .gitignore]: should we add this information in the readme?

**Comment by wonderingabout** [comment]
should we commit package-lock.json as well in this pr as dorus suggested in https://github.com/online-go/gtp2ogs/pull/260#discussion_r422624737?

File: docs/INSTALLATION-LINUX.md [hunk 28]
```diff
@@ -0,0 +1,59 @@
...
+ improvements and fixes. However devel is a "beta" branch which may be unstable
+ or contain bugs. If you prefer, you can install other more stable branches instead.
+
+note2: if you're a developer, you most likely would want to use your own fork rather
+ to be able to checkout easily and do other branch manipulations. To do that simply
+ replace `online-go` with your github name.
+
+gtp2ogs will be installed in `~/gtp2ogs/`.
+
+node_modules folder and package-lock.json are included in .gitignore
```
Comment (Dorus) [inline docs/INSTALLATION-LINUX.md:59]: package lock is not in gitigore right?

I'm not even sure on the relevance of this info here. It's pretty standard to have node_modules in gitignore.

**Comment by Dorus** [comment]
@roy7 This seems ok to me, can be merged when meta updates the final text i just mentioned.

File: docs/INSTALLATION-LINUX.md (hunk 28 above)
Comment (wonderingabout) [reply docs/INSTALLATION-LINUX.md]: yes you're right, it was my misunderstanding

I removed the gitignored of package-lock.json
and committed package-lock.json in this pr

**Comment by wonderingabout** [comment]
> @roy7 This seems ok to me, can be merged when meta updates the final text i just mentioned.

thanks, i did and added a note about license for developers, is it fine?
https://github.com/wonderingabout/gtp2ogs/blob/update-doc-node-local-install-nvm/docs/DEV.md#about-license

as we discussed yesterday on discord, some big projects have different licenses for different files, so i thought this precision is good to add

is it ok?

**Comment by Dorus** [comment]
@wonderingabout I'm not sure it's accurate that all files are under the same license. The code files at least are and should get a header for that.

**Comment by wonderingabout** [comment]
> @wonderingabout I'm not sure it's accurate that all files are under the same license. The code files at least are and should get a header for that.

@Dorus then should we delete the license information here and make a header in every file?
not something we would do in this PR then

**Comment by wonderingabout** [comment]
please do not merge yet, i need to rebase this in 1 commit with fixed commit notes

**Comment by wonderingabout** [comment]
rebased with no changes since the last few days, i fixed what @Dorus mentionned here https://github.com/online-go/gtp2ogs/pull/260#issuecomment-626346293, are there other things to fix or we can merge this?

File: connection.js [hunk 29]
```diff
@@ -173,6 +173,11 @@ class Connection {
                 return;
             }
 
+            // Don't connect if it is not our turn.
+            if (gamedata.player_to_move !== this.bot_id) {
```
Comment (Dorus) [inline connection.js:177]: Why revert this change we made recently?

File: connection.js [hunk 30]
```diff
@@ -305,7 +310,7 @@ class Connection {
         } else if (config.DEBUG) {
             console.log("There are no connected games");
         }
-        const connected_games_per_user = this.countGamesForPlayer(notification.user.id);
+        const connected_games_per_user = this.gamesForPlayer(notification.user.id);
```
Comment (Dorus) [inline connection.js:313]: Why revert this change we made recently?

File: connection.js [hunk 31]
```diff
@@ -500,10 +505,10 @@ class Connection {
             .catch(conn_log)
         }
     }
-    // processMove(gamedata) {
-    //     const game = this.connectToGame(gamedata.id)
-    //     game.makeMove(gamedata.move_number);
-    // }
+    processMove(gamedata) {
```
Comment (Dorus) [inline connection.js:508]: unused function, why outcomment?

File: connection.js [hunk 32]
```diff
@@ -535,7 +540,7 @@ class Connection {
             return;
         }
     }
-    countGamesForPlayer(player) {
+    gamesForPlayer(player) {
```
Comment (Dorus) [inline connection.js:543]: ?

File: pv.js [hunk 33]
```diff
@@ -12,6 +12,8 @@ class Pv {
             this.startupCheckSai = () => {}; // disable sai check for other bots.
         }
 
+        
+
```
Comment (Dorus) [inline pv.js:16]: Why add these newlines?

File: pv.js [hunk 34]
```diff
@@ -37,6 +39,9 @@ class Pv {
                          }[setting];
         this.CLPV =      { 'PHOENIXGO':  (/\([^()]*\)/g) }[setting];
     }
+    fetchOgspvType() {
```
Comment (Dorus) [inline pv.js:42]: Is this function used? Why does this pr add code?

File: connection.js (hunk 29 above)
Comment (wonderingabout) [reply connection.js]: its a rebasing last minute mistake
thankfully you spotted it

fixed now :+1:

File: connection.js (hunk 31 above)
Comment (wonderingabout) [reply connection.js]: fixed, same as https://github.com/online-go/gtp2ogs/pull/260#discussion_r425345678

File: connection.js (hunk 32 above)
Comment (wonderingabout) [reply connection.js]: fixed, same as https://github.com/online-go/gtp2ogs/pull/260#discussion_r425345678

File: pv.js (hunk 33 above)
Comment (wonderingabout) [reply pv.js]: fixed, same as https://github.com/online-go/gtp2ogs/pull/260#discussion_r425345678

File: pv.js (hunk 34 above)
Comment (wonderingabout) [reply pv.js]: fixed, same as https://github.com/online-go/gtp2ogs/pull/260#discussion_r425345678

**Comment by wonderingabout** [comment]
fixed all the last minute rebasing mistakes
thank you @Dorus for your vigilant eye, great! :100:  

should be ready to merge now :+1
and we can fix or enhance it in the future with this much better readme basis :+1:
